test/e2e/apimachinery/watchlist.go

```diff
@@ -52,7 +52,6 @@ var _ = SIGDescribe("API Streaming (aka. WatchList) [Serial] [Feature:WatchList]
 		secretInformer := cache.NewSharedIndexInformer(
 			&cache.ListWatch{
 				ListFunc: func(options metav1.ListOptions) (runtime.Object, error) {
-					framework.Fail("Unexpected list call")
```

Comment:
Can we instead switch to framework.Errorf or sth like that?

Reply:
I don't think so because `framework.ExpectNoError` calls `Fail` under the hood.

Reply:
Ideally if we could skip the test on jobs other than `kubernetes-e2e-gce-cos-alpha-features` but I am not aware of an easy way of doing it.

Reply:
Then let's set an error here and validate if it's non-nil at the end of the test instead.

Reply:
Why? The test will fail. We could change the error msg to be more descriptive.

Reply:
The secret informer fail to synchronise - [here](https://github.com/kubernetes/kubernetes/blob/a44ce346d7810465ceb15f421da972991b7b9245/test/e2e/apimachinery/watchlist.go#L79).

The reflector will fallback to `LIST` but it will get an error, in result the informer won't be able to sync the cache.

Reply:
Thanks.